lib/codec.js

```diff
@@ -241,53 +241,86 @@ function decodeFields(slice) {
         var tag = String.fromCharCode(slice[offset]); offset++;
         switch (tag) {
         case 'b':
+        // short-short-int = OCTET
             val = slice.readInt8(offset); offset++;
             break;
+        case 'B':
+        // short-short-uint = OCTET
+             val = slice.readUInt8(offset); offset++;
+             break;
         case 'S':
+        // long-string = long-uint *OCTET
             len = slice.readUInt32BE(offset); offset += 4;
             val = slice.toString('utf8', offset, offset + len);
             offset += len;
             break;
         case 'I':
+        // long-int = 4*OCTET
             val = slice.readInt32BE(offset); offset += 4;
             break;
+        case 'i':
+        // long-uint = 4*OCTET
+            val = slice.readUInt32BE(offset); offset += 4;
+            break;
         case 'D': // only positive decimals, apparently.
+        // decimal-value = scale long-uint
             var places = slice[offset]; offset++;
             var digits = slice.readUInt32BE(offset); offset += 4;
             val = {'!': 'decimal', value: {places: places, digits: digits}};
             break;
         case 'T':
+        // timestamp = long-long-uint; 64-bit POSIX
             val = ints.readUInt64BE(slice, offset); offset += 8;
             val = {'!': 'timestamp', value: val};
             break;
         case 'F':
+        // field-table = long-uint *field-value-pair
             len = slice.readUInt32BE(offset); offset += 4;
             val = decodeFields(slice.slice(offset, offset + len));
             offset += len;
             break;
         case 'A':
+        // field-array = long-int *field-value; array of values
             len = slice.readUInt32BE(offset); offset += 4;
             decodeArray(offset + len);
             // NB decodeArray will itself update offset and val
             break;
         case 'd':
+        // double = 8*OCTET; rfc1832 XDR double
             val = slice.readDoubleBE(offset); offset += 8;
             break;
         case 'f':
+        // float = 4*OCTET; IEEE-754
             val = slice.readFloatBE(offset); offset += 4;
             break;
         case 'l':
+        // long-long-uint = 8*OCTET
+            val = ints.readUInt64BE(slice, offset); offset += 8;
```

Comment:
According to https://www.rabbitmq.com/amqp-0-9-1-errata#section_3 'l' means a signed 64-bit integer for RabbitMQ, not an Unsigned 64-bit integer.